tests/test_physics/test_bargmann/test_bargmann_repr.py

```diff
@@ -0,0 +1,174 @@
+import numpy as np
+
+from mrmustard import math
+from mrmustard.lab import Attenuator, Coherent, Gaussian, Ggate
+from mrmustard.physics import representations
+from mrmustard.physics.bargmann import contract_two_Abc, reorder_abc, wigner_to_bargmann_rho
+from mrmustard.physics.representations import Bargmann
+from tests.random import random_Ggate, single_mode_unitary_gate, n_mode_mixed_state, Abc_triple
+from hypothesis import given
+
+
+def test_make_cat():
+    cat = Bargmann(*Coherent(1.0).bargmann()) + Bargmann(*Coherent(-1.0).bargmann())
+    assert np.allclose(cat.A[0], cat.A[1])
+    assert np.allclose(cat.b[0], -cat.b[1])
+
+
+def test_muldiv():
+    s1 = Bargmann(*Coherent(1.0).bargmann()) * Bargmann(*Coherent(-1.0).bargmann())
+    s2 = Bargmann(*Coherent(1.0).bargmann()) / Bargmann(*Coherent(-1.0).bargmann())
+    assert s1 is not None
+    assert s2 is not None
+
+
+def test_muldiv():
+    s1 = Bargmann(*Coherent(1.0).bargmann()) * 2.0
+    s2 = Bargmann(*Coherent(1.0).bargmann()) / 3.0
+    s3 = 4.0 * Bargmann(*Coherent(1.0).bargmann())
+    assert np.allclose(s1.c, Coherent(1.0).bargmann()[2] * 2.0)
+    assert np.allclose(s2.c, Coherent(1.0).bargmann()[2] / 3.0)
+    assert np.allclose(s3.c, Coherent(1.0).bargmann()[2] * 4.0)
+
+
+@given(Abc=Abc_triple(3))
+def test_reorder(Abc):
+    barg = Bargmann(*Abc)
+    barg.reorder((0, 2, 1))
+    assert np.allclose(barg.A[0], Abc[0][[0, 2, 1], :][:, [0, 2, 1]])
+    assert np.allclose(barg.b[0], Abc[1][[0, 2, 1]])
+
+
+@given(Abc=Abc_triple())
+def test_call(Abc):
+    """Test that we can call the PolyExpAnsatz object"""
+    A, b, c = Abc
+    barg = Bargmann(A, b, c)
+    assert np.allclose(barg(z=math.zeros_like(b)), c)
+
+
+def test_sub():
+    cat = Bargmann(*Coherent(1.0).bargmann()) - Bargmann(*Coherent(-1.0).bargmann())
+    assert np.allclose(cat.A[0], cat.A[1])
+    assert np.allclose(cat.b[0], -cat.b[1])
```

Comment:
Would it be `assert np.allclose(cat.b[0], cat.b[1])`? Since we use the sub on the second?

Reply:
no, because one is coeherent(1) and the other is coherent(-1)